mlir/include/mlir/Dialect/SCF/IR/SCF.h

```diff
@@ -107,6 +107,17 @@ LoopNest buildLoopNest(OpBuilder &builder, Location loc, ValueRange lbs,
                        function_ref<void(OpBuilder &, Location, ValueRange)>
                            bodyBuilder = nullptr);
 
+/// Perform a replacement of one iter OpOperand of an scf.for to the
+/// `replacement` value with a different type. A callback is used to insert
+/// cast ops inside the block to account for type differences.
+using ValueTypeCastFnTy =
+    std::function<Value(OpBuilder &, Location loc, Type, Value)>;
```

Comment:
I think this can be a `function_ref` rather than a `std::function` (which may heap alloc any captures), since helper does not store the function anywhere.  
```suggestion
using ValueTypeCastFnTy =
    llvm::function_ref<Value(OpBuilder &, Location loc, Type, Value)>;
```